extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java

```diff
@@ -304,16 +304,16 @@ private static Collection<DecoratorBuildItem> createRbacDecorators(String name,
         }
 
         // Add service account from extensions: use the one provided by the user always
-        String defaultServiceAccount = null;
+        Optional<String> defaultServiceAccount = config.getServiceAccount();
         String defaultServiceAccountNamespace = null;
         for (KubernetesServiceAccountBuildItem sa : serviceAccountsFromExtensions) {
             String saName = defaultIfEmpty(sa.getName(), name);
```

Comment:
The rest of the pull request is shifting away from the use of `defaultIfEmpty` to using `Optional` which is something that makes the code easier to read. 
Maybe we need to do the same here?

Reply:
Changed to `Optional.ofNullable(sa.getName()).orElse(name);` and removed `defaultIfEmpty` method.